### PR TITLE
[JENKINS-52075] Track the cumulative execution time of Pipeline jobs

### PR DESCRIPTION
**Pull request by stephenc**
See [JENKINS-52075](https://issues.jenkins-ci.org/browse/JENKINS-52075)

### CODE REVIEW

File: src/main/java/jenkins/metrics/impl/TimeInQueueAction.java [hunk 1]
```diff
@@ -113,18 +254,88 @@ public long getTotalDurationMillis() {
         return queuingDurationMillis + getBuildingDurationMillis();
     }
 
+    @Exported(visibility = 2)
+    public double getExecutorUtilization() {
+        long buildingDurationMillis = getBuildingDurationMillis();
+        return buildingDurationMillis > 0 ? (getExecutingTimeMillis() * 100 / buildingDurationMillis) / 100.0 : 1.0;
```
Comment (stephen-badger) [inline src/main/java/jenkins/metrics/impl/TimeInQueueAction.java]: You probably want to cast one of the operands to double to keep precision?

File: src/main/java/jenkins/metrics/impl/JenkinsMetricProviderImpl.java [hunk 2]
```diff
@@ -773,11 +825,33 @@ public boolean shouldSchedule(Queue.Task p, List<Action> actions) {
...
+                // we won the trim lottery
+                Queue queue = Jenkins.getInstance().getQueue();
+                // get the IDs that are currently known
+                Set<Long> currentIds = Stream.concat(Stream.of(queue.getItems()), queue.getLeftItems().stream())
+                        .map(Queue.Item::getId)
+                        .collect(Collectors.toCollection(HashSet::new));
+                // remove the currently known IDs from the previous list of known IDs
+                previousIds.removeAll(currentIds);
+                // now remove from the totals the ones that were previously known (at last trim) but are
+                // no longer known, we need to do it this was as otherwise any IDs that are added while
```
Comment (stephen-badger) [inline src/main/java/jenkins/metrics/impl/JenkinsMetricProviderImpl.java]: There is confusing comment grammar here. Also possibly the comment is incomplete?

File: src/main/java/jenkins/metrics/impl/TimeInQueueAction.java (hunk 1 above)
Comment (stephenc) [reply src/main/java/jenkins/metrics/impl/TimeInQueueAction.java]: no, i want to force it to 2 decimals ;-)

Comment (stephenc) [reply src/main/java/jenkins/metrics/impl/TimeInQueueAction.java]: but I should add a comment to that effect

**Pull request merged**